### ISSUE TITLE
Naming convention for tests

### ISSUE THREAD

**Issue by andrewchambers** [c-testsuite/c-testsuite#4]
Some tests are regression tests from compilers, but others are designed specifically for testing features. Is there a good naming system we could adopt.

**Comment by alexfru** [comment]
A few points...
- In my compiler I separated into different directories the tests that should compile (but don't) from the tests that shouldn't compile (but do); it's not quite about names, but improves organization
- I also made all dir/file names compatible with DOS since my compiler self-hosts there as well, so, short names (8 max for name, 3 max for extension), no fancy characters (just English letters, decimal digits and underscores)
- When you amass lots of tests in the same directory, descriptive names don't quite stand out anymore, you get distracted, so, it may be just as practical to move the description into the comment and reduce the name to a number or a prefix and a number
- If you want features mentioned unambiguously (and keep track of the standard), perhaps, you may mention particular standard clauses in some uniform way (e.g. C99 6.4.4.1 c 1-6); though, this is quite tedious, but then the question to ask would be, what's the purpose? Just another pile of random tests or something aiming at testing as much of the standard as possible? Don't rush answering unless it's a decided matter.

**Comment by andrewchambers** [comment]
> In my compiler I separated into different directories the tests that should compile (but don't) from the tests that shouldn't compile (but do); it's not quite about names, but improves organization

Agreed, naming by feature is less important than splitting by how to execute the test. 

> I also made all dir/file names compatible with DOS since my compiler self-hosts there as well, so, short names (8 max for name, 3 max for extension)

Technically a DOS specific runner can adapt the name before delegating to an emulator. But its good to keep the constraint in mind.

> this is quite tedious

Part of the motivations for this project is to isolate and share the *useful* tedious work across as many people as possible. So if it is worth it, but tedious, we can still attempt it.

**Comment by alexfru** [comment]
One problem with DOS is that you can't just dump a repo or extract an archive with arbitrary names to FAT. On top of that DOS lacks lots of simple tools to manipulate files or do scripting. Short and simple names would alleviate the issue in part. Running DOSBox may help as well (I think it'll make up DOS names from whatever's there).
